crates/isograph_schema/src/magic_mutation_fields.rs

```diff
@@ -496,3 +435,264 @@ fn skip_arguments_contained_in_field_map(
         processed_field_map_items,
     ))
 }
+
+#[derive(Debug)]
+struct GraphQLDirectiveDeserializer<'a> {
+    directive: &'a GraphQLDirective<ConstantValue>,
+}
+fn from_graph_ql_directive<'a, T: Deserialize<'a>>(
+    directive: &'a GraphQLDirective<ConstantValue>,
+) -> ProcessTypeDefinitionResult<T> {
+    T::deserialize(GraphQLDirectiveDeserializer { directive })
+        .map_err(|err| WithLocation::new(err, Location::generated()))
+}
+
+impl de::Error for ProcessTypeDefinitionError {
+    fn custom<T>(msg: T) -> Self
+    where
+        T: core::fmt::Display,
+    {
+        ProcessTypeDefinitionError::FailedToDeserialize(msg.to_string())
+    }
+}
+
+impl<'de> Deserializer<'de> for GraphQLDirectiveDeserializer<'de> {
+    type Error = ProcessTypeDefinitionError;
+
+    fn deserialize_any<V>(self, visitor: V) -> Result<V::Value, Self::Error>
+    where
+        V: de::Visitor<'de>,
+    {
+        visitor.visit_map(NameValuePairVec::new(&self.directive.arguments))
+    }
+
+    serde::forward_to_deserialize_any! {
+        bool i8 i16 i32 i64 i128 u8 u16 u32 u64 u128 f32 f64 char str string
+        bytes byte_buf option unit unit_struct newtype_struct seq tuple
+        tuple_struct map struct enum identifier ignored_any
+    }
+}
+
+struct NameValuePairVec<'a, T> {
```

Comment:
Can we name this `NameValuePairVecDeserializer` or something?

Reply:
Done here [`529d87c` (#31)](https://github.com/isographlabs/isograph/pull/31/commits/529d87c178cb724e986026b9e738c21c4db0a79e)